templates/tftmpl/variables.go

```diff
@@ -67,6 +67,12 @@ func newVariablesTF(w io.Writer, filename string, input *RootModuleInputData) er
 		}
 	}
 
+	if input.SourceInput != nil && input.SourceInput.SourceIncludesVariable() {
```

Comment:
SourceIncludesVariable() should always be true here so maybe I should just remove it from the check? Could also do a panic here if SourceIncludes() variable isn't true... this way we could catch programmer errors.

Reply:
Hm, I don't have a lot of experience in the conventions around panics in go, but I think removing the check would be good if it is always going to be true!

Reply:
I was thinking something like this:

```
	if input.SourceInput != nil {
		if input.SourceInput.SourceIncludesVariable() {
			if err = input.SourceInput.appendVariable(w); err != nil {
				return err
			}
		} else {
			// SourceIncludesVariable should always be true for a non-nil source input
			// if we got here, it means that a source_input's SourceIncludesVariable() function
			// was not setup correctly
			panic("SourceIncludesVariable is false for a non-nil source input")
		}
	}
```

I see panics as basically the equivalent of an assert, so for catching programmer errors.